react-instantsearch-hooks/geo-search/src/Airports.tsx

```diff
@@ -0,0 +1,86 @@
+import { useEffect, useRef } from 'react';
+import { renderToStaticMarkup } from 'react-dom/server';
+import { Marker, Popup, useMapEvents } from 'react-leaflet';
+import { DivIcon } from 'leaflet';
+import { useConnector, useSearchBox } from 'react-instantsearch-hooks-web';
+import connectGeoSearch, {
+  GeoHit,
+  GeoSearchConnectorParams,
+  GeoSearchWidgetDescription,
+} from 'instantsearch.js/es/connectors/geo-search/connectGeoSearch';
+
+type UseGeoSearchProps = GeoSearchConnectorParams;
+
+function useGeoSearch(props?: UseGeoSearchProps) {
+  return useConnector<GeoSearchConnectorParams, GeoSearchWidgetDescription>(
+    connectGeoSearch,
+    props
+  );
+}
+
+export function Airports() {
+  const skipViewEffect = useRef(false);
+
+  const { query, refine: refineQuery } = useSearchBox();
+  const {
+    items,
+    refine: refineItems,
+    currentRefinement,
+    clearMapRefinement,
+  } = useGeoSearch();
+
+  const onViewChange = ({ target }) => {
+    skipViewEffect.current = true;
+
+    if (query.length > 0) {
+      refineQuery('');
+    }
```

Comment:
if you only zoom, this also gets applied, however the searchbox doesn't clear, do you have any idea what happened?

Reply:
It is normally applied, is it still the case for you in the [latest preview](https://codesandbox.io/s/github/algolia/doc-code-samples/tree/feat/hooks-geo-search/react-instantsearch-hooks/geo-search) ?

Reply:
maybe here the input can be blurred first